examples/chatgpt/gpt_actions_library/gpt_action_canvaslms.ipynb

```diff
@@ -0,0 +1,651 @@
+    "Below are instructions on setting up authentication with this 3rd party application. Have questions? Check out [Getting Started Example](https://platform.openai.com/docs/getting-started) to see how this step works in more detail."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Pre-Action Steps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before you set up authentication in ChatGPT, please take the following steps in the application."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Canvas accounts : The first step is to ensure the canvas environment has users assigned to the appropriate role (student / teacher) with any access to appropriate courses\n",
+    "2. For testing purposes or service level account access create an approved integration access token\n",
+    "- Under Account -> Settings -> Approved Integrations\n",
+    "\n",
+    "<br> ![gptactions_canvaslms_settings.png](../../../images/gptactions_canvaslms_settings.png)\n",
+    "\n",
+    "- Create new Access Token\n",
+    "\n",
+    "<br> ![gptactions_canvaslms_integrationslist.png](../../../images/gptactions_canvaslms_integrationslist.png)\n",
+    "\n",
+    "- Provide purpose and expiration\n",
+    "\n",
+    "<br> ![gptactions_canvaslms_tokencreate.png](../../../images/gptactions_canvaslms_tokencreate.png)\n",
+    "\n",
+    "- Copy the new Token to use in the GPT Action (the token will not be accessible once you leave the page - ensure proper security measures for token access going forward)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### In ChatGPT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In ChatGPT, click on \"Authentication\" and choose **\"API Key\"**. Enter in the information below. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- **API Key**: Use the token from above \n",
+    "- **Auth Type**: Bearer\n",
+    "\n",
+    "<br> ![gptactions_canvaslms_auth.png](../../../images/gptactions_canvaslms_auth.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Post-Action Steps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once you've set up authentication in ChatGPT, follow the steps below in the application to test the Action."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- You will see a list of available actions from the API schema for Canvas (this schema above does not contain all possible API endpoints, be sure to edit the schema to produce the appropriate actions from canvas\n",
+    "- For OAuth implementation follow the guide here : https://canvas.instructure.com/doc/api/file.oauth.html - specific instructions can vary depending on the Canvas implementation. "
```

Comment:
Does that page have information on setting up the callback URL? that's one of the most challenging parts